src/model.ts

```diff
@@ -362,10 +363,14 @@ export class SpecQueryModelGroup {
     this._orderGroupBy = orderGroupBy;
   }
 
-  public getTopSpecQueryModel(): SpecQueryModel {
+  public isItemSpecQueryGroup(item: SpecQueryGroup<T> | T): item is SpecQueryGroup<T> {
```

Comment:
This looks more like static method? (Maybe just extract it outside of the class so it can be reused easily?)